plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergNodeLocalDynamicSplitPruning.java

```diff
@@ -0,0 +1,238 @@
+import static io.trino.orc.metadata.OrcType.createRootOrcType;
+import static io.trino.plugin.hive.HiveTestUtils.HDFS_ENVIRONMENT;
+import static io.trino.plugin.hive.HiveType.HIVE_INT;
+import static io.trino.plugin.hive.HiveType.HIVE_STRING;
+import static io.trino.plugin.iceberg.IcebergFileFormat.ORC;
+import static io.trino.spi.type.IntegerType.INTEGER;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static io.trino.type.InternalTypeManager.TESTING_TYPE_MANAGER;
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.apache.iceberg.types.Types.NestedField.optional;
+import static org.testng.Assert.assertEquals;
+
+public class TestIcebergNodeLocalDynamicSplitPruning
+{
+    private static final String ICEBERG_CATALOG_NAME = "iceberg";
+    private static final String SCHEMA_NAME = "test";
+    private static final String TABLE_NAME = "test";
+    private static final Column PARTITION_COLUMN = new Column("l_partkey", HIVE_INT, Optional.empty());
+    private static final ColumnIdentity PARTITION_COLUMN_IDENTITY = new ColumnIdentity(1, PARTITION_COLUMN.getName(), ColumnIdentity.TypeCategory.PRIMITIVE, ImmutableList.of());
+    private static final IcebergColumnHandle PARTITION_ICEBERG_COLUMN_HANDLE = new IcebergColumnHandle(PARTITION_COLUMN_IDENTITY, INTEGER, ImmutableList.of(), INTEGER, Optional.empty());
+    private static final Column DATA_COLUMN = new Column("col1", HIVE_STRING, Optional.empty());
+    private static final ColumnIdentity DATA_COLUMN_IDENTITY = new ColumnIdentity(2, DATA_COLUMN.getName(), ColumnIdentity.TypeCategory.PRIMITIVE, ImmutableList.of());
+    private static final Schema TABLE_SCHEMA = new Schema(
+            optional(PARTITION_COLUMN_IDENTITY.getId(), PARTITION_COLUMN.getName(), Types.IntegerType.get()),
+            optional(DATA_COLUMN_IDENTITY.getId(), DATA_COLUMN.getName(), Types.StringType.get()));
+    private static final OrcReaderConfig ORC_READER_CONFIG = new OrcReaderConfig();
+    private static final OrcWriterConfig ORC_WRITER_CONFIG = new OrcWriterConfig();
+    private static final ParquetReaderConfig PARQUET_READER_CONFIG = new ParquetReaderConfig();
+    private static final ParquetWriterConfig PARQUET_WRITER_CONFIG = new ParquetWriterConfig();
+
+    @Test
+    public void testDynamicSplitPruning()
+            throws IOException
+    {
+        IcebergConfig icebergConfig = new IcebergConfig();
+        HiveConfig hiveConfig = new HiveConfig();
+        HiveTransactionHandle transaction = new HiveTransactionHandle(false);
+        try (TempFile tempFile = new TempFile()) {
+            writeOrcContent(tempFile.file());
+
+            try (ConnectorPageSource emptyPageSource = createTestingPageSource(transaction, icebergConfig, hiveConfig, tempFile.file(), getDynamicFilter(getTupleDomainForPartitionSplitPruning()))) {
+                assertEquals(emptyPageSource.getClass(), EmptyPageSource.class);
```

Comment:
Instead of assertion on page source class, please assert on the output page which comes from calling `ConnectorPageSource#getNextPage`.